File: linked_list/singly_linked_list.cpp
#include<bits/stdc++.h>

using namespace std;

class SNode{
  public:
  int data;
  SNode *next;
  
  SNode(int info){
    this->data = info;
    this->next = nullptr;
  }
};

class SLList{
  public:
  SNode *head;
  SNode *tail;
  
  SLList(){
    this->head = nullptr;
    this->tail = nullptr;
  }
  
  void insert_node(int info){
    SNode* node = new SNode(info);
    if(!this=>head){
      this->head = node;
      else{
        this->tail->next= node;
      }
      this->tail = node;
};

void freeSLList(SNode* node){
  while(node){
    SNode* temp = node;
    node= node->next
    free(temp);
  }
}

File: linked_list/singly_linked_list.py
class SNode:
  def __init__(self, info):
    self.data = info
    self.next = None
    
class SLList:
  def __init__(self):
    self.head = None
    self.tail = None
    
   def insert(self, key):
    node = SNode(key)
    if not self.head:
      self.head = node
     else:
      self.tail.next = node
      self.tail = node
